mailpoet/tests/integration/Migrations/DbIndexesTest.php

```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Test\Config;
+
+use MailPoet\Config\Env;
+use MailPoetVendor\Doctrine\DBAL\Connection;
+
+class DbIndexesTest extends \MailPoetTest {
+  /**
+   * This test checks that we don't have an unique varchar index on a varchar column with a length bigger than 191.
+   * The length of 191 is the safe limit that works for all common MySQL configurations.
+   * We don't want to allow creating unique indexes on varchar columns with a length bigger than 191.
+   * @see https://stackoverflow.com/questions/15157227/mysql-varchar-index-length/16474039#16474039
+   */
+  public function testDbHasCorrectUniqueVarcharIndexes() {
+    $connection = $this->diContainer->get(Connection::class);
+    $incorrectIndexes = $connection->executeQuery("SELECT DISTINCT
+      ISS.TABLE_NAME,
+      ISS.INDEX_NAME,
+      ISS.COLUMN_NAME,
+      ISS.NON_UNIQUE,
+      ISC.CHARACTER_MAXIMUM_LENGTH,
+      ISC.DATA_TYPE
+    FROM INFORMATION_SCHEMA.STATISTICS ISS
+    JOIN INFORMATION_SCHEMA.COLUMNS ISC ON ISC.COLUMN_NAME = ISS.COLUMN_NAME AND ISC.TABLE_NAME = ISS.TABLE_NAME
+    WHERE ISS.TABLE_NAME LIKE :prefix
+      AND ISS.NON_UNIQUE = 0
```

Comment:
Just wondering why we need to care about `NON_UNIQUE` ?

Reply:
We actually care only about the UNIQUE ones (`NON_UNIQUE = 0`). As I understand, the problem is related only to unique indexes. MySQL needs the whole length to ensure uniqueness, but [in some cases, it can't use more than 191 characters and fails](https://stackoverflow.com/questions/15157227/mysql-varchar-index-length/16474039#16474039).